test/official.spec.ts

```diff
@@ -1,23 +1,187 @@
+    value:
+      'fffffffffffffffff800000000000000001bffffffffffffffffc8000000000000000045ffffffffffffffffc800000000000000001bfffffffffffffffff8000000000000000001',
+  },
+  { input: '10', value: '10' },
+  { input: '820001', value: '0001' },
+  {
+    input: 'C50583343434',
+    value: ['05', '343434'],
+  },
+  {
+    input: 'C601C402C203C0',
+    value: ['01', ['02', ['03', []]]],
+  },
+  {
+    input: 'C58083343434',
+    value: ['', '343434'],
+  },
+
+  {
+    input: 'C105',
+    value: ['05'],
+  },
+  {
+    input: 'C7C50583343434C0',
+    value: [['05', '343434'], []],
+  },
+  {
+    input: '83222222',
+    value: '222222',
+  },
+  {
+    input: 'C3010101',
+    value: ['01', '01', '01'],
+  },
+  {
+    input: 'C501C3C00000',
+    value: ['01', [[], '00', '00']],
+  },
+  {
+    input: 'C103',
+    value: ['03'],
+  },
+  {
+    input: 'C50102C20102',
+    value: ['01', '02', ['01', '02']],
+  },
+  {
+    input: 'C3010203',
+    value: ['01', '02', '03'],
+  },
+  {
+    input: 'C20102',
+    value: ['01', '02'],
+  },
+  {
+    input: 'C101',
+    value: ['01'],
+  },
+  {
+    input: 'C180',
+    value: [''],
+  },
+  {
+    input: 'C1C0',
+    value: [[]],
+  },
+  {
+    input: 'C103',
+    value: ['03'],
+  },
+
+  {
+    input: 'C2C103',
+    value: [['03']],
+  },
+  {
+    input: 'C20102',
+    value: ['01', '02'],
+  },
+  {
+    input: 'C3010203',
+    value: ['01', '02', '03'],
+  },
+  {
+    input: 'C401020304',
+    value: ['01', '02', '03', '04'],
+  },
+  {
+    input: 'C20180',
+    value: ['01', ''],
+  },
+  {
+    input: 'C50183010203',
+    value: ['01', '010203'],
+  },
+  { input: '82FFFF', value: 'ffff' },
+  { input: '07', value: '07' },
+  { input: '8180', value: '80' },
+  { input: 'C109', value: ['09'] },
+  { input: 'C58403030303', value: ['03030303'] },
+
+  { input: 'C3808005', value: ['', '', '05'] },
+  { input: 'C50183040404', value: ['01', '040404'] },
+  {
+    input: 'c330f9c030f93030ce3030303030303030bd303030303030',
+  },
+]
+
+function bufferArrayToStringArray(buffer: any): any {
+  return buffer.map((buf: any) => {
+    if (Array.isArray(buf)) {
+      return bufferArrayToStringArray(buf)
+    }
+    return buf.toString('hex')
+  })
+}
+
+describe('geth tests', function () {
+  for (const gethCase of gethCases) {
+    const buffer = Buffer.from(gethCase.input, 'hex')
+    it('should pass Geth test', function (done) {
+      try {
+        const output = RLP.decode(buffer)
+        if (Array.isArray(output)) {
+          const arrayOutput = bufferArrayToStringArray(output)
+          assert.equal(
+            JSON.stringify(arrayOutput),
+            JSON.stringify(gethCase.value!),
+            `invalid output: ${gethCase.input}`
+          )
+        } else {
+          assert.equal(output.toString('hex'), gethCase.value, `invalid output: ${gethCase.input}`)
```

Comment:
```suggestion
          assert.strictEqual(output.toString('hex'), gethCase.value, `invalid output: ${gethCase.input}`)
```

Update to non-deprecated `strictEqual`